File: Java/Console/Shared/src/shared/Response.java
package shared;

public class Response {

    String msg;
    String command;
    String data;

    public static Response parse(String msg) {
        String[] parts = msg.split(" ", 2);
        return new Response(parts[0], parts.length == 2 ? parts[1] : "");
    }

    Response(String command, String data) {
        this.command = command;
        this.data = data;
        this.msg = command + " " + data;
    }

    public String getMessage() {
        return msg;
    }

    public String getCommand() {
        return command;
    }

    public String getData() {
        return data;
    }
}

File: C#/Shared/Response.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Shared
{
    public class Response
    {

        public string cmd { get; private set; }
        public string data { get; private set; }
        public string message { get; private set; }

        public Response(string msg)
        {
            string[] split = msg.Split(new char[] { ' ' }, 2);

            message = msg;
            cmd = split[0];
            data = split.Length == 2 ? split[1] : null;
        }
    }
}
